packages/@react-spectrum/tag/src/Tag.tsx

```diff
@@ -10,82 +10,79 @@
-    isRemovable: removable,
-    isDisabled: disabled,
-    validationState: validationState !== undefined ? validationState : groupValidationState,
-    onRemove: props.onRemove || onRemove,
-    role
-  });
-  let {role: buttonRole, ...otherButtonProps} = clearButtonProps;
-  let icon = props.icon || (isInvalid && <Alert />);
+    isRemovable,
+    isDisabled,
+    item,
+    onRemove
+  }, state);
+
+  return (
+    <div
+      {...otherProps}
+      {...styleProps}
+      {...mergeProps(tagProps, hoverProps, focusProps, labelProps)}
+      role={tagProps.role}
+      className={classNames(
+        styles,
+        'spectrum-Tags-item',
+        {
+          'is-disabled': isDisabled,
+          'focus-ring': isFocusVisible,
+          'is-focused': isFocusVisible,
+          'spectrum-Tags-item--removable': isRemovable,
+          'is-hovered': isHovered
+        },
+        styleProps.className
+      )}>
+      <SlotProvider
+        slots={{
+          icon: {UNSAFE_className: classNames(styles, 'spectrum-Tag-icon'), size: 'XS'},
+          text: {UNSAFE_className: classNames(styles, 'spectrum-Tag-content')},
+          tagRemoveButton: {UNSAFE_className: classNames(styles, 'spectrum-Tag-action')}
+        }}>
+        {typeof props.children === 'string' ? <Text><span role="gridcell">{props.children}</span></Text> : props.children}
+        {isRemovable && <TagRemoveButton item={item} state={state} {...clearButtonProps} />}
+      </SlotProvider>
+    </div>
+  );
+}
+
+function TagRemoveButton(props) {
+  props = useSlotProps(props, 'tagRemoveButton');
+  let {styleProps} = useStyleProps(props);
+  let clearBtnRef = useRef();
 
   return (
-    <FocusRing focusRingClass={classNames(styles, 'focus-ring')}>
-      <div
-        {...styleProps}
-        {...mergeProps(tagProps, hoverProps)}
-        className={classNames(
-          styles,
-          'spectrum-Tags-item',
-          {
-            'is-disabled': disabled,
-            // 'is-selected': isSelected,
-            'spectrum-Tags-item--removable': removable,
-            'is-invalid': isInvalid,
-            'is-hovered': isHovered
-          },
-          styleProps.className
-        )}>
-        {icon && React.cloneElement(icon, {size: 'S', UNSAFE_className: classNames(styles, 'spectrum-Tags-itemIcon')})}
-        <span
-          {...labelProps}
-          className={classNames(styles, 'spectrum-Tags-itemLabel')}>
-          {props.children}
-        </span>
-        {removable &&
-          <span role={buttonRole}>
-            <ClearButton
-              tabIndex={tagProps.tabIndex}
-              focusClassName={classNames(styles, 'is-focused')}
-              UNSAFE_className={classNames(styles, 'spectrum-Tags-itemClearButton')}
-              {...otherButtonProps} />
-          </span>
-        }
-      </div>
-    </FocusRing>
+    <span
+      {...styleProps}
+      ref={clearBtnRef}>
+      <ClearButton
+        focusClassName={classNames(styles, 'focus-ring')}
```

Comment:
The focus ring for the ClearButton is a bit off center vertically and feels a bit thin to me, couldn't find a XD file that showed what it should look like though so this might be ok for now:
![image](https://user-images.githubusercontent.com/9661127/140235405-2575f864-cb60-4488-af2b-60beeff7b9c0.png)

Reply:
updated in latest push. although i have a tag padding vs button padding problem so the 'x' is still off center. basically if i have both tag padding and 'x' button padding, the right side of the tag is extra wide. i would love to center the 'x', but i think my css skills are still a little lacking in how to achieve that. let me know if you have any suggestions

Reply:
Ah, I was actually fine with the positioning of the "x" button previously, was mainly concerned about the positioning of the blue ring around it. IMO you can revert back to the old styling and ignore the "x" button's focus ring positioning for now since we are still talking about if it should even be focusable